runtime/compiler/env/J9ObjectModel.cpp

```diff
@@ -406,15 +406,19 @@ J9::ObjectModel::maxContiguousArraySizeInBytes()
 bool
 J9::ObjectModel::isDiscontiguousArray(int32_t sizeInBytes)
    {
-   if (sizeInBytes > TR::Compiler->om.maxContiguousArraySizeInBytes())
+   /* When off heap allocation is enabled both large and small arrays are
```

Comment:
Going through these changes again, I don't think this needs to be updated for off heap. It just checks if the array size is less than or equal to the max contiguous array size and having off heap enabled shouldn't matter for this. I probably made this change for `getAddressOfElement` [1]. There should have been a separate case for off heap instead of adding it under `!TR::Compiler->om.isDiscontiguousArray(comp, objectPointer)` if block. 

@rmnattas do you mind doing a test run without this commit and [1] hoisted in a if block before [2].


[1] https://github.com/VermaSh/openj9/blob/c8ab9fd55f26ae92bc4bc38367ba2ce30eb7ea95/runtime/compiler/env/J9ObjectModel.cpp#L693-L699 "getAddressOfElement"
[2] https://github.com/VermaSh/openj9/blob/c8ab9fd55f26ae92bc4bc38367ba2ce30eb7ea95/runtime/compiler/env/J9ObjectModel.cpp#L691 "if block"

Reply:
This would remove the need for the later commit https://github.com/rmnattas/openj9/commit/a6d94dbd43
Without the "Update isDiscontiguousArray" commit?

If what I'm understanding is correct then caller needs to make sure to do a separate off-heap check before calling `isDiscontiguousArray` and that doesn't seem like a good way to make it. Besides if we do the separate off-heap check or not, `isDiscontiguousArray` needs to know about off-heap.

Reply:
We can keep both:
1- https://github.com/eclipse-openj9/openj9/pull/19228/commits/741023b196bc363ac19b5412b012d308a1e4bd60
2- https://github.com/rmnattas/openj9/commit/a6d94dbd43

And maybe update 2 to move the off-heap case outside the `isDiscontiguousArray` check

Reply:
>This would remove the need for the later commit https://github.com/rmnattas/openj9/commit/a6d94dbd43
Without the "Update isDiscontiguousArray" commit?

We still need that change although we'll need to move the off-heap case outside of the `isDiscontiguousArray` check block.

> caller needs to make sure to do a separate off-heap check before calling isDiscontiguousArray and that doesn't seem like a good way to make it.

The caller will need to do a separate check anyways. `true` from `isDiscontiguousArray` if off-heap is disabled means that the data portion is right after the header whereas if off-heap is enabled `true` doesn't mean anything since the array can be on heap or off heap. If we revert [741023b](https://github.com/eclipse-openj9/openj9/commit/741023b196bc363ac19b5412b012d308a1e4bd60) we can be certain that it will always return true for heap allocated contiguous arrays.

Reply:
I see, this goes back to what we mean when we say "DiscontiguousArray", is it array-elements being discontiguous or array-header + arrayElements are discontiguous, it seems the latter meaning is better, former can be checked by array-lets checks. But its have been used interchangeably.

I'll remove that commit from this PR, and do the testing and fix the other commit for the next PR. So this can be set for merge.